tests/integration_tests/performance/test_huge_pages.py

```diff
@@ -139,3 +139,45 @@ def test_negative_huge_pages_plus_balloon(uvm_plain):
         match="Machine config error: Firecracker's huge pages support is incompatible with memory ballooning.",
     ):
         uvm_plain.basic_config(huge_pages=HugePagesConfig.HUGETLBFS_2MB)
+
+
+@pytest.mark.skipif(
+    global_props.host_linux_version == "4.14",
+    reason="MFD_HUGETLB | MFD_ALLOW_SEALING only supported on kernels >= 4.16",
+)
+def test_negative_huge_pages_plus_initrd(uvm_with_initrd):
+    """Tests that huge pages and initrd cannot be used together"""
+    uvm_with_initrd.jailer.daemonize = False
+    uvm_with_initrd.spawn()
+    uvm_with_initrd.memory_monitor = None
+
+    # Ensure setting huge pages and then telling FC to boot an initrd does not work
+    with pytest.raises(
+        RuntimeError,
+        match="Boot source error: Firecracker's huge pages support is incompatible with initrds.",
+    ):
+        # `basic_config` first does a PUT to /machine-config, which will apply the huge pages configuration,
+        # and then a PUT to /boot-source, which will register the initrd
+        uvm_with_initrd.basic_config(
+            boot_args="console=ttyS0 reboot=k panic=1 pci=off",
+            use_initrd=True,
+            huge_pages=HugePagesConfig.HUGETLBFS_2MB,
+            add_root_device=False,
+            vcpu_count=1,
+        )
+
+    # Ensure telling FC about the initrd first and then setting huge pages doesn't work
```

Comment:
can be a separate test too.

Reply:
Please see https://github.com/firecracker-microvm/firecracker/pull/4360#discussion_r1475829061

Reply:
My argument for having separate tests is we would exclude (potentially unknown) residual effects from previous operations, so all tests are "clean", but I agree the returns are diminishing at some point, so up to you.